### PR TITLE
mon: OSDMonitor: fix the check error of pg creating

### PR DESCRIPTION
**Pull request by jordan41177**
Fix: http://tracker.ceph.com/issues/17169

Signed-off-by: DesmondS <desmond.s@inwinstack.com>

### CODE REVIEW

**Comment by jordan41177** [comment]
Fix: http://tracker.ceph.com/issues/17169

**Comment by tchaikov** [comment]
@jordan41177 the change looks good to me, could you add a simple test in `test_mon_osd_pool()` in `qa/workunits/cephtool/test.sh` ?

File: qa/workunits/cephtool/test.sh [hunk 1]
```diff
@@ -1514,6 +1514,9 @@ function test_mon_osd_pool_set()
   wait_for_clean
   ceph osd pool set $TEST_POOL_GETSET pgp_num 10
 
+  ceph osd pool set $TEST_POOL_GETSET pg_num 106
```
Comment (tchaikov) [inline qa/workunits/cephtool/test.sh]: could you squash this commit into the previous one? or add a prefix ("qa/workunits/cephtool/test.sh: ") to the title of the commit message of this commit?

Comment (jordan41177) [inline qa/workunits/cephtool/test.sh]: Okay @tchaikov , I will fix it after a week...(for a trip). Sorry for late.

**Comment by jordan41177** [comment]
@tchaikov  
It's done. I combined two changes in one commit.

**Comment by tchaikov** [comment]
@jordan41177 could you put "Fixes: http://tracker.ceph.com/issues/17169" in your commit message? right before your "Signed-off-by" line.

**Comment by hjwsm1989** [comment]
lgtm

**Comment by jordan41177** [comment]
@tchaikov  @hjwsm1989 
how about this PR?

**Comment by tchaikov** [comment]
tested at http://pulpito.ceph.com/kchai-2016-10-09_12:58:06-rados-wip-kefu-testing---basic-mira/

failed tests are tracked at
- http://tracker.ceph.com/issues/16943
- http://tracker.ceph.com/issues/17540

**Pull request merged**